src/alignment.jl

```diff
@@ -100,41 +104,74 @@ function Base.show(io::IO, aln::Alignment)
       print(io, "  CIGAR string: ", cigar(aln))
 end
 
-"""
-    seq2ref(aln::Alignment, i::Integer)::Tuple{Int,Operation}
-
-Map a position `i` from sequence to reference.
-"""
-function seq2ref(aln::Alignment, i::Integer)::Tuple{Int,Operation}
-    idx = findanchor(aln, i, Val{true})
+# generic function for mapping between sequence, reference and alignment positions
+# getsrc specifies anchor source position getter
+# getdest specifies anchor destination position getter
+function pos2pos(aln::Alignment, i::Integer,
+                 srcpos::Function, destpos::Function)::Tuple{Int,Operation}
+    idx = findanchor(aln, i, srcpos)
     if idx == 0
-        throw(ArgumentError("invalid sequence position: $i"))
+        if srcpos === seqpos
+            throw(ArgumentError("invalid sequence position: $i"))
+        elseif srcpos === refpos
+            throw(ArgumentError("invalid reference position: $i"))
+        elseif srcpos === agnpos
+            throw(ArgumentError("invalid alignment position: $i"))
+        else
+            throw(ArgumentError("Unknown position getter: $srcpos"))
+        end
     end
     anchor = aln.anchors[idx]
-    refpos = anchor.refpos
-    if ismatchop(anchor.op)
-        refpos += i - anchor.seqpos
+    pos = destpos(anchor)
+    if ismatchop(anchor.op) ||
+        ((srcpos === agnpos) && ((destpos === seqpos) && isinsertop(anchor.op) || (destpos === refpos) && isdeleteop(anchor.op))) ||
+        ((destpos === agnpos) && ((srcpos === seqpos) && isinsertop(anchor.op) || (srcpos === refpos) && isdeleteop(anchor.op)))
+        pos += i - srcpos(anchor)
```

Comment:
Might be better to use `==` here - if the type of any of the integers change, this code will needlessly break.